roles/openshift_autoheal/templates/objects.yml.j2

```diff
@@ -0,0 +1,136 @@
+apiVersion: v1
```

Comment:
Can we use the same template we have in the autoheal repo itself? https://github.com/openshift/autoheal/blob/master/template.yml

Reply:
Sure, although maintaining it in two places is unfortunate, that might be the reality until there's an operator managing this thing. The ansible role will have to be the supported/canonical template in the meantime, I think.

Reply:
Currently this could be an OpenShift template, but it will be problematic in the future if we want to create objects outside of the `openshift-autoheal` namespace, as templates can't create objects outside the namespace where they are applied: the `namespace` field of objects is just overwritten.

So, for example, if in the future we need to create a role binding inside the `kube-system` namespace then we won't be able to use one template, we will need at least two templates, one for each namespace.

On the other hand the list of objects generated with Jinja2 is completely flexible, it can create objects in any namespace.

Note that isn't just theoretical, I already hit this issue during an experiment that I am doing to add an API server that allows us to store things persistently in etcd: I need to add a binding for the `extension-apiserver-authentication-reader` role, which lives in the `kube-system` namespace.

I'd prefer to stick to the Jinja2 template for this reason, but I will change if that is unacceptable. @ironcladlou let me know.

Reply:
By the way, we don't need to wait for the operator to have the template only on one place. When we have this merged we can change the `README.md` of the openshift/authoeal so that it refers to this new role, and then we can remove the template there.

Reply:
>Currently this could be an OpenShift template, but it will be problematic in the future if we want to create objects outside of the openshift-autoheal namespace, as templates can't create objects outside the namespace where they are applied: the namespace field of objects is just overwritten.

>So, for example, if in the future we need to create a role binding inside the kube-system namespace then we won't be able to use one template, we will need at least two templates, one for each namespace.

If the role itself is the atomic unit of work, and individual tasks are idempotent, atomicity is preserved across multiple `oc apply` operations. Do you agree that multiple template files could be applied separately against multiple namespaces to achieve the goal in your example?

I only bring this up because my understanding is that all new installable components should have a general focus on divestiture from Ansible insofar as manifests and configuration (at a minimum) are concerned: the architectural direction of the product is to use Go operators to manage these components' installations and upgrades, and in that model, the resources need to be OpenShift Templates (or any other core) resource serializations.